service-deployment/src/grafana/grafana_config.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+chown -R grafana:grafana /var/lib/grafana /var/log/grafana
+
+# Script to configure grafana datasources and dashboards.
+# Intended to be run before grafana entrypoint...
+# ENTRYPOINT [\"/run.sh\"]"
+
+GRAFANA_URL=${GRAFANA_URL}
```

Comment:
>GRAFANA_URL [](start = 14, length = 11)

where to get grafana url?

Reply:
I used to set the env at dockerfile, Yundong suggests to set the env in yaml file when deploying.

Reply:
that is a good approach. can you list all env. variables in the beginning of the file so people know where them come from.

---
In reply to: [157503386](https://github.com/Microsoft/pai/pull/102#discussion_r157503386) [](ancestors = 157503386)